solana/pyth2wormhole/client/src/main.rs

```diff
@@ -303,56 +260,119 @@ async fn handle_attest(
         rpc_interval,
     ));
 
-    let message_q_mtx = Arc::new(Mutex::new(P2WMessageQueue::new(
-        Duration::from_millis(attestation_cfg.min_msg_reuse_interval_ms),
-        attestation_cfg.max_msg_accounts as usize,
-    )));
-
-    // Create attestation scheduling routines; see attestation_sched_job() for details
-    let mut attestation_sched_futs = batches.into_iter().map(|(batch_no, batch)| {
-        attestation_sched_job(
-            batch,
-            batch_no,
-            batch_count,
-            n_retries,
-            retry_interval,
-            daemon,
-            rpc_cfg.clone(),
-            p2w_addr,
-            config.clone(),
-            Keypair::from_bytes(&payer.to_bytes()).unwrap(),
-            message_q_mtx.clone(),
-        )
-    });
-
-    info!("Spinning up attestation sched jobs");
-
-    let results = futures::future::join_all(attestation_sched_futs).await; // May never finish for daemon mode
+    // This loop governs mapping account reloads in daemon mode. It is
+    // broken if daemon != true at the and
+    loop {
+        let config = get_config_account(&lock_and_make_rpc(&rpc_cfg).await, &p2w_addr).await?;
+
+        // Use the mapping if specified
+        if let Some(mapping_addr) = attestation_cfg.mapping_addr.as_ref() {
```

Comment:
I suggest keeping around the original attestation_cfg (pass it immutably into the function), then make a copy here and add any new symbols from the mapping to the copy. That way the configuration for each iteration of the loop is not dependent on previous iterations.